app/src/test/java/io/appium/uiautomator2/model/settings/AbstractSettingTests.java

```diff
@@ -20,68 +20,127 @@
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.ExpectedException;
 import org.mockito.Spy;
 
 import io.appium.uiautomator2.common.exceptions.UiAutomator2Exception;
 
+import static org.mockito.ArgumentMatchers.anyInt;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.verify;
 import static org.mockito.MockitoAnnotations.initMocks;
 
 public class AbstractSettingTests {
-
-    @Rule
-    public ExpectedException thrown = ExpectedException.none();
+    @Spy
+    private DummyIntegerSetting dummyIntegerSetting;
 
     @Spy
-    private DummySetting dummySetting;
+    private DummyLongSetting dummyLongSetting;
 
     @Before
     public void setUp() {
         initMocks(this);
     }
 
     @Test
-    public void shouldThrowExceptionIfTypeIsNotValid() {
-        thrown.expect(UiAutomator2Exception.class);
-        thrown.expectMessage("Invalid setting value type. Got: java.lang.String. Expected: java.lang.Long");
-        dummySetting.update("test");
+    public void dummyIntegerSettingShouldThrowExceptionIfTypeIsNotValid() {
+        try {
+            dummyIntegerSetting.update("test");
+        } catch (UiAutomator2Exception e) {
+            Assert.assertEquals("Invalid setting value type. Got: java.lang.String. Expected: java.lang.Integer", e.getMessage());
+        }
+    }
+
+    @Test
```

Comment:
you could use `@Test(expect=Exception.class)`

Reply:
I did that originally but _Codacy_ was not happy as it wanted at lead one `assert` | `fail` in the method body.

Reply:
yep, codacy could sometimes be stupid about tests :(